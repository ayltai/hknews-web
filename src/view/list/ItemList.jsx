import React from 'react';
import { List } from 'react-admin';

import { CardList } from './CardList';
import { ItemFilter } from './ItemFilter';

export class ItemList extends React.PureComponent {
    componentDidUpdate = () => {
        window.scrollTo(0, 0);
    }

    render() {
        return (
            <List
                bulkActionButtons={false}
                exporter={false}
                filters={<ItemFilter />}
                title='ra.page.list'
                {...this.props}>
                <CardList />
            </List>
        );
    }
}
