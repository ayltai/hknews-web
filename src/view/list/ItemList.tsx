import React from 'react';

import { CardList } from './CardList';
import { ItemFilter } from './ItemFilter';

const { List } = require('react-admin');

export class ItemList extends React.PureComponent<{}> {
    public constructor(props : any) {
        super(props);
    }

    public componentDidUpdate : () => void = () : void => {
        window.scrollTo(0, 0);
    }

    public render() : React.ReactElement {
        return (
            <List
                bulkActionButtons={false}
                exporter={false}
                filters={<ItemFilter />}
                {...this.props}>
                <CardList />
            </List>
        );
    }
}
