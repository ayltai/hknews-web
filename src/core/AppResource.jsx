import React from 'react';
import { Resource } from 'react-admin';

import { ItemShow } from '../view/detail/ItemShow';
import { ItemList } from '../view/list/ItemList';

export const AppResource = props => (
    <Resource
        name={props.name}
        icon={props.icon}
        list={ItemList}
        show={ItemShow}
        options={{
            label : props.name
        }}
        {...props} />
);
