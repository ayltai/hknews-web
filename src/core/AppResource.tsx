import React from 'react';

import { ItemShow } from '../view/detail/ItemShow';
import { ItemList } from '../view/list/ItemList';

const { Resource } = require('react-admin');

interface IProps {
    name : string;
    icon : any;
}

export const AppResource : React.FunctionComponent<IProps> = (props : IProps) : React.ReactElement => (
    <Resource
        name={props.name}
        list={ItemList}
        show={ItemShow}
        icon={props.icon}
        {...props} />
);
