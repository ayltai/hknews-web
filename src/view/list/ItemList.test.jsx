import { createMuiTheme, MuiThemeProvider, } from '@material-ui/core';
import { render, } from 'enzyme';
import React from 'react';
import { TestContext, } from 'react-admin';
import renderer from 'react-test-renderer';

import { ItemList, } from './ItemList';

const defaultStoreForList = {
    admin : {
        resources : {
            foo : {
                list : {
                    ids         : [],
                    params      : {},
                    selectedIds : [],
                    total       : 0,
                },
            },
        },
    },
};

describe('<ItemList />', () => {
    it('renders without errors', () => {
        expect(renderer.create(
            <TestContext initialState={defaultStoreForList}>
                <MuiThemeProvider theme={createMuiTheme({
                    palette : {
                        type : 'dark',
                    },
                })}>
                    <ItemList
                        basePath='/'
                        ids={[]}
                        location={{
                            pathname : '',
                        }}
                        resource='foo'>
                        <div />
                    </ItemList>
                </MuiThemeProvider>
            </TestContext>)
            .toJSON())
            .toMatchSnapshot();
    });

    it('mount without errors', () => {
        render(
            <TestContext initialState={defaultStoreForList}>
                <MuiThemeProvider theme={createMuiTheme({
                    palette : {
                        type : 'dark',
                    },
                })}>
                    <ItemList
                        basePath='/'
                        ids={[]}
                        location={{
                            pathname : '',
                        }}
                        resource='foo'>
                        <div />
                    </ItemList>
                </MuiThemeProvider>
            </TestContext>
        );
    });
});

