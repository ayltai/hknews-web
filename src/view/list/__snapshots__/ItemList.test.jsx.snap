// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemList /> renders without errors 1`] = `
<div
  className="list-page RaList-root-1"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular RaListToolbar-toolbar-7 MuiToolbar-gutters"
  >
    <form
      className="RaFilter-form-14 RaFilterForm-form-15"
      onSubmit={[Function]}
    >
      <div
        className="filter-field RaFilterFormInput-body-17"
        data-source="source"
      >
        <div
          className="MuiFormControl-root MuiFormControl-marginDense"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root RaLabeled-label-20 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense"
            data-shrink={true}
            htmlFor="source"
          >
            <span>
              labels.publisher
            </span>
          </label>
          <div
            className="RaLabeled-value-21"
          >
            <div
              className="MuiLinearProgress-root MuiLinearProgress-colorPrimary RaLinearProgress-root-46 MuiLinearProgress-indeterminate"
              input={
                Object {
                  "checked": undefined,
                  "name": "source",
                  "onBlur": [Function],
                  "onChange": [Function],
                  "onFocus": [Function],
                  "value": "",
                }
              }
              resource="foo"
              role="progressbar"
            >
              <div
                className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
                style={Object {}}
              />
              <div
                className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
                style={Object {}}
              />
            </div>
          </div>
        </div>
        <div
          className="RaFilterFormInput-spacer-18"
        >
          Â 
        </div>
      </div>
      <div
        className="RaFilterForm-clearFix-16"
      />
    </form>
    <span />
    <div
      className="MuiToolbar-root MuiToolbar-regular RaTopToolbar-root-68 RaListToolbar-actions-8 MuiToolbar-gutters"
    />
  </div>
  <div
    className="RaList-main-2"
  >
    <div
      className="MuiPaper-root MuiCard-root RaList-content-3 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        style={
          Object {
            "flexGrow": 1,
            "padding": 16,
          }
        }
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        />
      </div>
      <div
        className="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
      />
    </div>
  </div>
</div>
`;
